### DISCUSSION TITLE
Redirect www subdomain to apex domain

### DISCUSSION THREAD

**Discussion by natke** [community/community#23351] [Other Feature Feedback, Questions, & Ideas]
<p>Referencing this <a href="https://github.com/isaacs/github/issues/1675" rel="noopener nofollow ugc">issue</a> where <strong><a href="https://github.com/yoannchaudet" rel="noopener nofollow ugc">yoannchaudet</a></strong> said to post over here, we are experiencing the issue described and need guidance on our GitHub pages config and DNS record config. Thanks!</p>
<p>The GitHub pages website is <a href="http://onnxruntime.github.io" rel="noopener nofollow ugc">onnxruntime.github.io</a>, with apex domain onnxruntime.ai and www subdomain.</p>
<p>If there is a doc describing this, happy to follow that.</p>
<p>Thank you!</p>

**Comment by yoannchaudet** [comment]
<p>Hello!</p>
<p>We document how to secure both an apex domain and the www variant <a href="https://docs.github.com/en/pages/configuring-a-custom-domain-for-your-github-pages-site/managing-a-custom-domain-for-your-github-pages-site#configuring-an-apex-domain-and-the-www-subdomain-variant">here</a>.</p>
<p>In your case you simply need to change your custom domain to <code>onnxruntime.ai</code> instead of <code>www.onnxruntime.ai</code>.</p>
<p>Let me know if that helps!</p>

**Comment by natke** [comment]
<p>Thank you for the quick reply!</p>
<p>We are now getting a persistent TLS cert error. These are the <a href="https://who.is/dns/onnxruntime.ai" rel="noopener nofollow ugc">DNS records</a>. Any idea what is wrong here?</p>
<p>Thanks again!</p>

**Comment by yoannchaudet** [comment]
<p>I pulled logs for your specific certificate and can see that all requests to provision the certificate have failed consistently.</p>
<p>I think this is because your domain is pinning a single certificate authority:</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/e/a/ea9fbc9c2cf9b969f923f48adc3308f91c99a70e.png" data-download-href="/uploads/short-url/xtzXFThwcHRcMfGn46ysyU2k1JA.png?dl=1" title="image"><img src="https://user-images.githubusercontent.com/14911070/181076470-95369ce6-d82b-48b7-be8e-f439521a2a1e.png" alt="image" data-base62-sha1="xtzXFThwcHRcMfGn46ysyU2k1JA" width="690" height="348" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/e/a/ea9fbc9c2cf9b969f923f48adc3308f91c99a70e_2_690x348.png, https://aws1.discourse-cdn.com/github/optimized/3X/e/a/ea9fbc9c2cf9b969f923f48adc3308f91c99a70e_2_1035x522.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/e/a/ea9fbc9c2cf9b969f923f48adc3308f91c99a70e.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/e/a/ea9fbc9c2cf9b969f923f48adc3308f91c99a70e_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">1330×672 120 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>GitHub Pages is using Let’s Encrypt so you will either need to remove the restriction you have in place (i.e. remove the <code>CAA</code> record) or add Let’s Encrypt to the list (i.e. add an extra <code>CAA</code> record with value  <code>0 issue "letsencrypt.org"</code>). We briefly describe this in our <a href="https://docs.github.com/en/pages/configuring-a-custom-domain-for-your-github-pages-site/troubleshooting-custom-domains-and-github-pages#https-errors">documentation</a> and you can learn more about <a href="https://letsencrypt.org/docs/caa/"><code>CAA</code> on Let’s Encrypt’s website directly</a>.</p>
<p>I’ll make a note for us to improve the detection.</p>
<p>Once your DNS is adjusted your certificate should provision just fine. It may take few hours though since the few background retries that are happening automatically now are counting toward Let’s Encrypt’s per-domain rate limits.</p>

**Comment by natke** [comment]
<p>Awesome, thank you for that! I’ve applied to get that record removed. Hopefully will resolve things. Thanks again for the speedy response, and yes any extra diagnostics would be very helpful</p>

**Comment by yoannchaudet** [comment]
<p>I can see the DNS change propagated and your certificate has been issued :tada:</p>
